src/cdk-cli-integ-tests.ts

```diff
@@ -0,0 +1,292 @@
+import { Component, github, javascript } from 'projen';
+
+export interface CdkCliIntegTestsWorkflowProps {
+  /**
+   * Runners for the workflow
+   */
+  readonly buildRunsOn: string;
+
+  /**
+   * Runners for the workflow
+   */
+  readonly testRunsOn: string;
+
+  /**
+   * GitHub environment name for approvals
+   *
+   * MUST be configured to require manual approval.
+   */
+  readonly approvalEnvironment: string;
+
+  /**
+   * GitHub environment name for running the tests
+   *
+   * MUST be configured without approvals, and with the following vars and secrets:
+   *
+   * - vars: AWS_ROLE_TO_ASSUME_FOR_TESTING
+   *
+   * And the role needs to be configured to allow the AssumeRole operation.
+   */
+  readonly testEnvironment: string;
+
+  /**
+   * Packages that are locally transfered (we will never use the upstream versions)
+   */
+  readonly localPackages: string[];
+}
+
+/**
+ * Add a workflow for running the tests
+ *
+ * This MUST be a separate workflow that runs in privileged context. We have a couple
+ * of options:
+ *
+ * - `workflow_run`: we can trigger a privileged workflow run after the unprivileged
+ *   `pull_request` workflow finishes and reuse its output artifacts. The
+ *   problem is that the second run is disconnected from the PR so we would need
+ *   to script in visibility for approvals and success (by posting comments, for
+ *   example)
+ * - Use only a `pull_request_target` workflow on the PR: this either would run
+ *   a privileged workflow on any user code submission (might be fine given the
+ *   workflow's `permissions`, but I'm sure this will make our security team uneasy
+ *   anyway), OR this would mean any build needs human confirmation which means slow
+ *   feedback.
+ * - Use a `pull_request` for a regular fast-feedback build, and a separate
+ *   `pull_request_target` for the integ tests. This means we're building twice.
+ *
+ * Ultimately, our build isn't heavy enough to put in a lot of effort deduping
+ * it, so we'll go with the simplest solution which is the last one: 2
+ * independent workflows.
+ *
+ * projen doesn't make it easy to copy the relevant parts of the 'build' workflow,
+ * so they're unfortunately duplicated here.
+ */
+export class CdkCliIntegTestsWorkflow extends Component {
+  constructor(repo: javascript.NodeProject, props: CdkCliIntegTestsWorkflowProps) {
+    super(repo);
+
+    const buildWorkflow = repo.buildWorkflow;
+    const runTestsWorkflow = repo.github?.addWorkflow('build-and-integ');
+    if (!buildWorkflow || !runTestsWorkflow) {
+      throw new Error('Expected build and run tests workflow');
+    }
+    ((buildWorkflow as any).workflow as github.GithubWorkflow);
+
+    runTestsWorkflow.on({
+      pullRequestTarget: {
+        branches: ['main'],
+      },
+    });
+    // The 'build' part runs on the 'integ-approval' environment, which requires
```

Comment:
Why did you decide to do it like this? Seems counter intuitive because the `integ` part is the sensitive one which we want to approve before running, no?